react/features/base/dialog/components/native/ConfirmDialog.js

```diff
@@ -1,107 +1,171 @@
+        this._onConfirm = this._onConfirm.bind(this);
     }
 
     _onCancel: () => void;
 
     /**
-     * Renders the 'No' button.
+     * Callback to cancel button.
      *
-     * NOTE: The {@code ConfirmDialog} is the only dialog right now that
-     * renders 2 buttons, mainly for clarity.
+     * @private
+     * @returns {void}
+     */
+    _onCancel() {
+        const { onCancel } = this.props;
+
+        onCancel && onCancel();
+
+        this.setState({
+            visible: !this.state.visible
+        });
+    }
+
+    _onConfirm: () => void;
+
+    /**
+     * Callback for the confirm button.
      *
-     * @inheritdoc
+     * @private
+     * @returns {void}.
+     */
+    _onConfirm() {
+        const { onSubmit } = this.props;
+
+        onSubmit && onSubmit();
+
+        this.setState({
+            visible: !this.state.visible
+        });
+    }
+
+    /**
+     * Renders the dialog description.
+     *
+     * @returns {React$Component}
      */
-    _renderAdditionalButtons() {
-        const { _dialogStyles, cancelKey, onDecline, t } = this.props;
+    _renderDescription() {
+        const { children, descriptionKey, t } = this.props;
+        const description
+            = typeof descriptionKey === 'string'
+                ? t(descriptionKey)
+                : this._renderHTML(
+                    t(descriptionKey?.key, descriptionKey?.params)
+                );
 
         return (
-            <TouchableOpacity
-                onPress = { onDecline || this._onCancel }
-                style = { [
-                    _dialogStyles.button,
-                    brandedDialog.buttonFarLeft,
-                    _dialogStyles.buttonSeparator
-                ] }>
-                <Text style = { _dialogStyles.buttonLabel }>
-                    { t(cancelKey || 'dialog.confirmNo') }
+            <>
+                <Text style = { styles.dialogDescription }>
+                    { description }
+                </Text>
+                <Text style = { styles.dialogDescription }>
+                    { children }
                 </Text>
-            </TouchableOpacity>
+            </>
         );
     }
 
     /**
-     * Implements {@code BaseSubmitDialog._renderSubmittable}.
+     * Implements {@code Component#render}.
      *
      * @inheritdoc
      */
-    _renderSubmittable() {
-        if (this.props.children) {
-            return this.props.children;
-        }
-
-        const { _dialogStyles, contentKey, t } = this.props;
-        const content
-            = typeof contentKey === 'string'
-                ? t(contentKey)
-                : this._renderHTML(t(contentKey.key, contentKey.params));
+    render() {
+        const { cancelLabel, confirmLabel, t } = this.props;
+        const { visible } = this.state;
 
         return (
-            <Text style = { _dialogStyles.text }>
-                { content }
-            </Text>
+            <View>
```

Comment:
Why do we need to wrap this in a view? Isn't it ok to just have the container be the outermost component?

Reply:
This is what the dependency documentation recommends, but I can try removing it to see if it affects something or not.

Reply:
If it's recommended, let's do it :-)